### PR TITLE
LLS: AbsSystem and AbsSurvey

### PR DESCRIPTION
**Pull request by profxj**
Subclasses for AbsSystem and AbsSurvey
docs
tests
notebooks

a number of utils changes too.

### CODE REVIEW

File: linetools/isgm/lls.py [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+        # Combine
+        tau_model = tau_LL + tau_Lyman
+
+        # Kludge around the limit
+        pix_LL = np.argmin(np.fabs( wv_rest- 911.3*u.AA))
+        pix_kludge = np.where((wv_rest > 911.5*u.AA) & (wv_rest < 912.8*u.AA))[0]
+        tau_model[pix_kludge] = tau_model[pix_LL]
+        
+        # Fill in flux
+        model = copy.deepcopy(spec)
```
Comment (nhmc) [inline linetools/isgm/lls.py]: Use spec.copy() instead. Then you can also remove the copy import.

File: linetools/isgm/lls.py [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+        """ Measure zpeak from an ionic transition
+        """
+        if self.ions is None:
+            print('get_zpeak: Need to fill ions with get_ions first.')
+            return
+
+        # Ions for analysis
+        low_ions = [ (14,2), (6,2), (13,2), (26,2), (13,3)]
+        high_ions= [(14,4), (6,4)]
+
```
Comment (nhmc) [inline linetools/isgm/lls.py]: Comment here which ions these are  (i.e. NV, OVI etc)

File: linetools/isgm/lls.py [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+            # Skip primordial LLS
+            print('lls.zpeak: No transition in {:s}'.format(self.clm_analy.clm_fil))
+            return (0,0), 0.
+        # Return
+        return ion, vpeak
+
+    # Output
+    def __repr__(self):
+        return ('[{:s}: {:s} {:s}, zabs={:g}, NHI={:g}, tau_LL={:g}, [M/H]={:g} dex]'.format(
+                self.__class__.__name__,
```
Comment (nhmc) [inline linetools/isgm/lls.py]: I prefer [Z/H] to [M/H] (because of X, Y, Z notation for metals), but not a big deal.

**Comment by nhmc** [comment]
Not sure if you're finished with this, but I left some minor comments.

It would be nice to move all the `clm` names to `N`, where possible.

File: linetools/isgm/lls.py (hunk 1 above)
Comment (profxj) [inline linetools/isgm/lls.py]: Done.

File: linetools/isgm/lls.py (hunk 2 above)
Comment (profxj) [inline linetools/isgm/lls.py]: done

File: linetools/isgm/lls.py (hunk 3 above)
Comment (profxj) [inline linetools/isgm/lls.py]: Done.

I begin to forget why I used "M" so long ago..

**Comment by profxj** [comment]
Changes committed.

**Comment by nhmc** [comment]
Looks good to me, please merge when tests pass.

**Comment by profxj** [comment]
Merging.

**Pull request merged**